### ISSUE TITLE
resetApp leads to "chrome not reachable" error on hybrid Android app

### ISSUE THREAD

**Issue by dekelev** [appium/appium#6564]
## The problem

I have a test that starts with the _resetApp_ command and after the app is restarted, I'm getting the "chrome not reachable " error.

Same issue when using _closeApp().launchApp()_ instead of _resetApp()_

I'm using the "autoWebview: true" and "noReset: true" capabilities.
I've also tried not to use the _autoWebview_ capability and added a manual connect to the first WebView after app starts - didn't work, even with _sleep_ period before the first attempt to connect to the Cordova WebView.

The same test works for Cordova on iOS app (same app) and iOS native app with WebView.

Android hybrid app also suffers from this issue after using _resetApp()_ or _closeApp().launchApp()_ and seems to behave better when using the new capability _recreateChromeDriverSessions_ set to _true_. This capability did NOT solved the issue for Cordova on Android.
## Environment
- Appium version (or git revision) that exhibits the issue: 1.4.16, 1.5.2, 1.5.3
- Last Appium version that did not exhibit the issue (if applicable): it used to work, but I can't remember which version I used
- Desktop OS/version used to run Appium: Windows 8.1
- Node.js version (unless using Appium.app|exe): 4.4.5
- Mobile platform/version under test: Android versions: 4.4.4, 5.0, 5.0.1, 5.1.1
- Real device or emulator/simulator: real device
- Appium CLI or Appium.app|exe: Appium CLI
## Details

```
Error response status: 100,  Selenium error: chrome not reachable
  (Session info: webview=33.0.0.0)
  (Driver info: chromedriver=2.18.343845 (73dd713ba7fbfb73cbb514e62641d8c96a94682a),platform=Windows NT 6.3 x86_64)
    at exports.newError (D:\dvlp\Github\MobileAutoTests\tests\node_modules\wd\lib\utils.js:139:13)
    at callbackWithDataBase (D:\dvlp\Github\MobileAutoTests\tests\node_modules\wd\lib\callbacks.js:75:19)
    at D:\dvlp\Github\MobileAutoTests\tests\node_modules\wd\lib\webdriver.js:174:5
    at Request._callback (D:\dvlp\Github\MobileAutoTests\tests\node_modules\wd\lib\http-utils.js:87:7)
    at Request.init.self.callback (D:\dvlp\Github\MobileAutoTests\tests\node_modules\wd\node_modules\request\request.js:368:22)
    at Request.<anonymous> (D:\dvlp\Github\MobileAutoTests\tests\node_modules\wd\node_modules\request\request.js:1219:14)
    at IncomingMessage.<anonymous> (D:\dvlp\Github\MobileAutoTests\tests\node_modules\wd\node_modules\request\request.js:1167:12)
    at endReadableNT (_stream_readable.js:913:12)
```
## Link to Appium logs

https://gist.github.com/dekelev/2953f849cb0f492781086e9fc1cad34a (Android app using a WebView)
https://gist.github.com/dekelev/1117f5b679d2e90f3c5cf292d9f7bf6d (Cordova on Android using a second WebView)
## Code To Reproduce Issue

```
return driver.resetApp() // resetting app to solve issue on Android, where Appium loses control on WebViews after using them in a previous test
            .resolve().then(() => {
                return driver.waitForElementById('exampleElement')
```

**Comment by dekelev** [comment]
I've found a workaround:

Set the _recreateChromeDriverSessions_ capability to _true_.
Make sure that the _autoWebview_ capability is set to _true_.

Add the following steps right after _resetApp_ (or after _closeApp().launchApp()_):

1. connect to the native context (using _context('NATIVE_APP')_)
2. connect to the first WebView context (Cordova main WebView)

That's it!

**Comment by guilhotarras** [comment]
I am having this same issue after a resetApp() on Android. The problem is that on iOS, the same test works with success. So in my opinion this is a bug because we have different behavior on Android/iOS for the same function.
Can we reopen this ?

**Comment by lukhjh1** [comment]
I can confirm this works. I reopen the app using `adb shell am start -n packageName/activityName` and following  @dpgraham suggestions I managed to avoid the "chrome not reachable" error.
Appium 1.2.1 (latest build as of Sep 15 2017)

**Comment by kramnadroj** [comment]
Sorry if there's an issue with bumping old issues, but in case anyone was having the same problem, I solved mine by changing the context to NATIVE_APP _before_ calling driver.resetApp. The app I'm working on launches with a native view but then after that page, nothing but webview. I had no issue until I tried to run a second scenario where I call resetApp and change my context back to native.

driver.getcontext showed that I was on NATIVE_APP but I still got the chrome not reachable when trying to select a native elelment.

I'm not sure why, but it seemed to be because I called resetApp while still in WEBVIEW, changing to NATIVE_APP _before_ reset fixed it and I had no issues selecting native elements again
